### ISSUE TITLE
Add orientation to cylindrical Observables

### ISSUE THREAD

**Issue by jngrad** [espressomd/espresso#4023]
Depends on #4019.

Observables based on cylindrical coordinates have a `center` and `axis` parameters, but no `orientation`, which is required to get a well-defined phi angle when the `axis` parameter is not aligned with the x, y, z axes.

For performance reasons, when looping over particles, pre-compute the `phi0` angle with the `orientation` vector once, and use the `phi0` variable instead of the `orientation` vector in the for loop.

**Comment by jngrad** [comment]
> For performance reasons, when looping over particles, pre-compute the `phi0` angle with the `orientation` vector once, and use the `phi0` variable instead of the `orientation` vector in the for loop.

Actually, this should be measured first. If there is no performance degradation when using `orientation` in the for loop, then there is no need to pre-calculate the `phi0` value.

**Comment by christophlohrmann** [comment]
I could do this on the coding day

**Comment by jngrad** [comment]
@christophlohrmann you can create a new PR starting from my branch in #4042, leaving 90d21a8551 out. The python tests fail because there is no `orientation` argument in the python interface yet. Also for backward compatibility, we could use a random orientation vector in the case where no `orientation` vector is provided and the longitudinal axis is collinear to the x-axis.

**Comment by RudolfWeeber** [comment]
>Also for backward compatibility, we could use a random orientation vector in the case where no orientation vector is provided and the longitudinal axis is >collinear to the x-axis.

I’m against introducing random vectors for backward compatibility

The interface is allowed to break non-silently for 4.2 in important cases.

Users should be forced to specify the phi=0 direction for cylindrical historgrams. If they don’t care, they can easily pick one of the Cartesian axes.

 

Could you please also insert a check throwing a runtime error, if someone specifies a phi=0 direction which is not orthogonal to the cylindrical aies?
